generators/chipyard/src/main/scala/HarnessBinders.scala

```diff
@@ -240,6 +240,34 @@ class WithSimSerial extends OverrideHarnessBinder({
   }
 })
 
+class WithSimTLSerial(withHarnessSerialAdapter: Boolean = false) extends OverrideHarnessBinder({
+  (system: CanHavePeripheryTLSerial, th: HasHarnessSignalReferences, ports: Seq[ClockedIO[SerialIO]]) => {
+    implicit val p = chipyard.iobinders.GetSystemParameters(system)
+    ports.map({ port =>
+      withClockAndReset(port.clock, th.harnessReset) {
+        val lRam = LazyModule(new SerialRAM(
+          p(SerialTLKey).get.width,
+          p(SerialTLKey).get.memParams.master.size,
+          p(SerialTLKey).get.memParams.master.base,
+          managerEdge = system.serdesser.get.managerNode.edges.in(0),
+          clientEdge = system.serdesser.get.clientNode.edges.out(0)
+        ))
+        val ram = Module(lRam.module)
+        ram.io.ser <> port.bits
+
+        require(lRam.serdesser.module.mergedParams == system.serdesser.get.module.mergedParams,
+          "Mismatch between chip-side diplomatic params and testram diplomatic params")
+        if (withHarnessSerialAdapter) {
+          val success = SerialAdapter.connectSimSerial(Some(ram.io.tsi_ser), port.clock, th.harnessReset.asBool)
```

Comment:
It seems like configs that include both this and `WithSIMTSISerial` will instantiate two `SerialAdapter`s. Not sure if that will cause problems in simulation. I think they will be hooked up correctly but I don't think one of them will be pruned and I'm kind of worried about the VPI functions getting called multiple times.